README.md

```diff
@@ -14,18 +14,55 @@ Implement the tool `st` that does exact pattern matching using a suffix tree. Te
 # Report
 
 ## Specify if you have used a linear time or quadratic time algorithm.
+We ended up with a quadratic time algorithm, we played around with mccreight, but other projects got the better of us, and ate what little time we had left.
 
 ## Insights you may have had while implementing and comparing the algorithms.
+Our implementation choice on how to handle leaves (in this case they're just ints) sparked a lengthy yet very enlightening discussion on how we interpret and envision abstract data structure in our minds eye, turns out we each have our differences.  
+
+Packing all our functions into the classes might not be the smartes way of doing this (having the bft as a method under SuffixTree make the usage of the method a bit unintuitive.), but it made importing nice and easy.
 
 ## Problems encountered if any.
+Other than getting one step closer to enlightenment, most problems we encountered were -as usual- temporary strokes of idiocy on our part (indexing -once again- is hard).
+Most difficulties we had were in early implemetation choices for our Knæ class, eg. do we want a dictionary or list implementation, and how do we handled it. How much information do we need in each Knæ, and how do we deal with our choice of ints as leaves. 
 
 ## Correctness
 
 *Describe experiments that verifies the correctness of your implementations.*
 
+We could not come up with a good idea for testing just the construction, but if the tree is constructed correctly the search should output the same matches as the naive implementation of exact pattern matching. 
+We generated test data and compared outputs from our ST implementation with the naive pattern matching, to verifiy correctness. Furthermore we also tried with empty strings, just in case something went wierd. For palindromes, markov, random and single letter, the output seemed correct. biggest problem was that it was not necessarily in the same order, but we don't care about that.
+
+
 ## Running time
 
 *Describe experiments that verifies that your implementation of `st` uses no more time than O(n) or O(n²) (depending on the algorithm) for constructing the suffix tree and no more than O(m) for searching for a given read in it. Remember to explain your choice of test data. What are “best” and “worst” case inputs?*
 
 *If you have graphs that show the running time--you probably should have--you can embed them here like we did in the previous project.*
 
+We generated 2 series of data for the running time, one series using increasing N (input string length), at three different choices of M (pattern length), the other keeping N constant and varying M. The relation between lines in the resulting plots should reveal the time complexity.
+
+When running the code on strings only consisting of 'A's with varying lengths of sequences we see a clear O(n^2) time complexity for the contruction of trees.
```

Comment:
Do you? I always find it bloody hard to tell if a curve is $O(n^2)$ or, e.g. $O(n^{2.5})$. Fitting to a second-degree polynomial does fuck all since you can fit polynomials to anything.

I would generally recommend (and I have several times in this class) that you plot the running time divided by the expected running time. That should converge to a line if you have the right running time, and it is pretty easy to see if you have a line!